darts/tests/test_torch_forecasting_model.py

```diff
@@ -0,0 +1,142 @@
+            # no exception and no warning is raised
+
+        def test_create_instance_existing_model_with_name_force_fit_no_reset(self):
+            model_name = 'test_model'
+            model1 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name)
+            # no exception is raised
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series, epochs=1)
+
+            with self.assertWarnsRegex(UserWarning, "You already have model data for the '{}' name and you "
+                                                    "initialized it with".format(model_name)):
+                model2 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name, force=True)
+
+            with self.assertRaisesRegex(ValueError, "You forced initialization of the model but.*"):
+                model2.fit(series, epochs=1)
+
+        def test_create_instance_existing_model_with_name_force_fit_with_reset(self):
+            model_name = 'test_model'
+            model1 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name)
+            # no exception is raised
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series, epochs=1)
+
+            model2 = RNNModel(10, 10, work_dir=self.temp_work_dir, model_name=model_name, force=True)
+            model2.reset_model()
+            model2.fit(series, epochs=1)
+            # no exception is raised
+
+        # n_epochs=20, fit|epochs=None, total_epochs=0 - train for 20 epochs
+        def test_train_from_0_to_20_no_fit_epochs(self):
+            model1 = RNNModel(10, 10, n_epochs=20, work_dir=self.temp_work_dir)
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series)
+
+            self.assertEqual(model1.total_epochs, 20)
+
+        # n_epochs = 20, fit|epochs=None, total_epochs=20 - train for another 20 epochs
+        def test_train_from_20_to_40_no_fit_epochs(self):
+            model1 = RNNModel(10, 10, n_epochs=20, work_dir=self.temp_work_dir)
+
+            times = pd.date_range('20130101', '20130410')
+            pd_series = pd.Series(range(100), index=times)
+            series = TimeSeries.from_series(pd_series)
+            model1.fit(series)
+            self.assertEqual(model1.total_epochs, 20)
+
+            model1.fit(series)
+            self.assertEqual(model1.total_epochs, 40)
+
+        # n_epochs = 20, fit|epochs=None, total_epochs=10 - train for another 20 epochs
+        def test_train_from_10_to_20_no_fit_epochs(self):
```

Comment:
```suggestion
        def test_train_from_10_to_30_no_fit_epochs(self):
```?

Reply:
Also, why call it '...no_fit...'?

Reply:
because training is based on the `n_epochs` parameter instead of `fit`'s `epochs` param. So it's rather `...no_fit_epochs`

Reply:
Ah I see, thanks for explaining! But it should still say `...10_to_30...`, no?

Reply:
the second number tells the value of `n_epochs` parameter, but you're right, this is somewhat confusing. I'll think how to better convey the test purpose.

Reply:
fixed the naming convention.